toplevel/coqtop.ml

```diff
@@ -469,34 +472,33 @@ let init_toplevel arglist =
         let iload_path = build_load_path opts in
         let require_libs = require_libs opts in
         let stm_options = opts.stm_flags in
-        try
-          let open Vernac.State in
-          let doc, sid =
-            Stm.(new_doc
-                   { doc_type = Interactive opts.toplevel_name;
-                     iload_path; require_libs; stm_options;
-                   }) in
-          let state = { doc; sid; proof = None; time = opts.time } in
-          Some (load_init_vernaculars opts ~state), opts
-        with any -> flush_all(); fatal_error_exn any
+        let open Vernac.State in
+        let doc, sid =
+          Stm.(new_doc
+                 { doc_type = Interactive opts.toplevel_name;
+                   iload_path; require_libs; stm_options;
+                 }) in
+        let state = { doc; sid; proof = None; time = opts.time } in
+        Some (load_init_vernaculars opts ~state), opts
       (* Non interactive: we perform a sequence of compilation steps *)
       end else begin
```

Comment:
I think this is the part that has to be stripped out and that by the way has been causing quite a few other bugs; initialization is merged with a few different modes including vio and compilation, so no wonder the control flow has a lot to be desired; modes in the toplevel should be made clearer.